### PR TITLE
Fix remove_dbg

### LINKED ISSUE #5129: `Remove dbg!` assist removes more text than it should.
Applied to a code like `dbg!(a + b).foo()`, turns it into `a + b.foo()`.

### PR DESCRIPTION
**Pull request by petr-tik**
Closes #5129 

Addresses two issues:
- keep the parens from dbg!() in case the call is chained or there is
semantic difference if parens are excluded
- Exclude the semicolon after the dbg!(); by checking if it was
accidentally included in the macro_call

investigated, but decided against:
fix ast::MacroCall extraction to never include semicolons at the end -
this logic lives in rowan.

Defensively shorten the macro_range if there is a semicolon token.
Deleted unneccessary temp variable macro_args

Renamed macro_content to "paste_instead_of_dbg", because it isn't a
simple extraction of text inside dbg!() anymore

### CODE REVIEW

File: crates/ra_assists/src/handlers/remove_dbg.rs [hunk 1]
```diff
@@ -126,4 +140,52 @@ fn foo(n: usize) {
             "dbg!(n.checked_sub(4))",
         );
     }
+
+    #[test]
+    fn test_remove_dbg_keep_semicolon() {
+        // https://github.com/rust-analyzer/rust-analyzer/issues/5129#issuecomment-651399779
```
Comment (petr-tik) [inline crates/ra_assists/src/handlers/remove_dbg.rs:146]: @flodiebold is this what you were referring to in [your comment](https://github.com/rust-analyzer/rust-analyzer/issues/5129#issuecomment-651399779)? I couldn't reproduce the with a comment above the dbg!, but this test showed an error that I fixed

File: crates/ra_assists/src/handlers/remove_dbg.rs [hunk 2]
```diff
@@ -126,4 +140,52 @@ fn foo(n: usize) {
...
+        // not quite though
+        // adding a comment at the end of the line makes
+        // the ast::MacroCall to include the semicolon at the end
+        let code = "
+let res = <|>dbg!(1 * 20); // needless comment
+";
+        let expected = "
+let res = 1 * 20; // needless comment
+";
+        check_assist(remove_dbg, code, expected);
```
Comment (matklad) [inline crates/ra_assists/src/handlers/remove_dbg.rs:156]: Let's inlice code & expected variables, and also make them raw string. That way, rust-analyzer will syntax-highlighth them, and it's the style we use for most assits tests.

File: crates/ra_assists/src/handlers/remove_dbg.rs [hunk 3]
```diff
@@ -27,19 +27,32 @@ pub(crate) fn remove_dbg(acc: &mut Assists, ctx: &AssistContext) -> Option<()> {
         return None;
     }
 
-    let macro_range = macro_call.syntax().text_range();
+    let semicolon_on_end = macro_call.semicolon_token().is_some();
+    let is_leaf = macro_call.syntax().next_sibling().is_none();
 
-    let macro_content = {
-        let macro_args = macro_call.token_tree()?.syntax().clone();
+    let macro_end = match semicolon_on_end {
```
Comment (matklad) [inline crates/ra_assists/src/handlers/remove_dbg.rs:33]: Please use an `if` here and elsewhere

File: crates/ra_assists/src/handlers/remove_dbg.rs [hunk 4]
```diff
@@ -27,19 +27,32 @@ pub(crate) fn remove_dbg(acc: &mut Assists, ctx: &AssistContext) -> Option<()> {
...
+    let macro_range = TextRange::new(macro_call.syntax().text_range().start(), macro_end);
+    let paste_instead_of_dbg = {
+        let text = macro_call.token_tree()?.syntax().text();
+
+        // leafines determines if we should include the parenthesis or not
+        let slice_index: TextRange = match is_leaf {
+            // leaf means - we can extract the contents of the dbg! in text
+            true => TextRange::new(TextSize::of('('), text.len() - TextSize::of(')')),
+            // not leaf - means we should keep the parens
+            false => TextRange::new(TextSize::from(0 as u32), text.len()),
```
Comment (matklad) [inline crates/ra_assists/src/handlers/remove_dbg.rs:48]:
```suggestion
            false => TextRange::up_to(text.len()),
```

**Comment by matklad** [comment]
bors r+

Thanks!

**Pull request merged**